src/main/java/de/medizininformatikinitiative/torch/service/CrtdlProcessingService.java

```diff
@@ -125,7 +130,7 @@ private Mono<Void> processIntern(AnnotatedCrtdl crtdl, String jobID, Flux<Patien
         logger.debug("Process patient batches with a concurrency of {}", maxConcurrency);
         return preProcessedCoreBundle.flatMapMany(coreSnapshot ->
                 batches
-                        .flatMap(batch -> crtdl.consentKey()
+                        .flatMap(batch -> crtdl.consentKey(consentContexts)
```

Comment:
please extract a method here and use the empty set to differentiate. also you are executing `Mono.just(PatientBatchWithConsent.fromBatch(batch))` in every case now. That should not be done an existing consent context